### PR TITLE
Css modules

### PR DESCRIPTION
**Pull request by nranas**
Migration to CSS Modules and fixing #7 with support for "data-disable-default-styling" attribute.

### CODE REVIEW

File: src/app.js [hunk 1]
```diff
@@ -1,16 +1,21 @@
...
 import { executeAction, update } from './api'
 import state from './state'
+import styles from './app.css'
 
 export default class App extends Component {
-  constructor() {
+  constructor({ defaultStyling }) {
     super()
     this.state = state.get()
+    this.className = `ct-app`
```
Comment (hfjallemark) [inline src/app.js]: You can use normal `'` here instead of the string interpolation ```.

File: src/components/json-debugger.js [hunk 2]
```diff
@@ -1,26 +1,15 @@
...
 export default ({ resource }) => {
+  let className = `${styles.root} ct-json-debugger`
+
+  if (process.env.NODE_ENV == 'production') {
+    className += ` ${styles.hide}`
+  }
+
   return (
-    <pre className={`${styles.root} ct-json-debugger`}>
+    <pre className={`${className}`}>
```
Comment (hfjallemark) [inline src/components/json-debugger.js]: You can do: `<pre className={className}>`.

File: src/components/elements/section.css [hunk 3]
```diff
@@ -0,0 +1,18 @@
+.root {
+  margin: 5px 0;
+}
+
+.nested {
+  background: rgba(0 0 0 0.03);
```
Comment (hfjallemark) [inline src/components/elements/section.css]: Commas missing.

File: src/components/elements/section.css [hunk 4]
```diff
@@ -0,0 +1,18 @@
...
+  border: solid 1px rgba(0 0 0 0.03);
```
Comment (hfjallemark) [inline src/components/elements/section.css]: Commas missing.

File: src/components/elements/inputs/checkbox.js [hunk 5]
```diff
@@ -1,4 +1,4 @@
-import styles from './input-styles'
+import styles from './input-styles.css'
```
Comment (hfjallemark) [inline src/components/elements/inputs/checkbox.js]: Rename `input-styles.css` to `input.css`.

File: src/components/document.css [hunk 6]
```diff
@@ -0,0 +1,18 @@
...
+.header {
+  align-items: center;
+  border-bottom: 1px solid #eee;
+  display: flex;
+  margin-bottom: 30px;
+  padding: 10px 0 20px;
+}
+
+.headerText {
+  flex: 1px;
```
Comment (hfjallemark) [inline src/components/document.css]: Should be `flex: 1`.

File: src/components/elements/link.css [hunk 7]
```diff
@@ -0,0 +1,13 @@
+.root {
+  color: #5DB0EA;
+  font-size: 15px;
+  font-style: normal;
+  font-weight: 300;
+  padding: 5px 0;
+  text-decoration: none
```
Comment (hfjallemark) [inline src/components/elements/link.css]: Missing semicolon.

File: src/components/error-message.css [hunk 8]
```diff
@@ -0,0 +1,13 @@
+.root {
+  background: #FF6767;
+  border: solid 1px #DC6363;
+  border-bottom: 0;
+  border-radius: 3px 3px 0 0;
+  box-shadow: inset 0 -2px 2px rgba(0 0 0 0.05px);
```
Comment (hfjallemark) [inline src/components/error-message.css]: Commas missing.

Comment (nranas) [inline src/components/error-message.css]: Fixed

File: src/components/elements/link.css (hunk 7 above)
Comment (nranas) [inline src/components/elements/link.css]: Fixed

File: src/components/elements/inputs/checkbox.js (hunk 5 above)
Comment (nranas) [inline src/components/elements/inputs/checkbox.js]: Fixed

File: src/app.js (hunk 1 above)
Comment (nranas) [inline src/app.js]: Fixed

File: src/components/json-debugger.js (hunk 2 above)
Comment (nranas) [inline src/components/json-debugger.js]: Fixed

File: src/components/elements/section.css (hunk 3 above)
Comment (nranas) [inline src/components/elements/section.css]: Fixed

File: src/components/elements/section.css (hunk 4 above)
Comment (nranas) [inline src/components/elements/section.css]: Fixed

File: src/components/document.css (hunk 6 above)
Comment (nranas) [inline src/components/document.css]: Fixed

**Pull request merged**